docs/understanding-your-pipeline/canonical-event/index.md

```diff
@@ -36,274 +42,490 @@ The Snowplow data table is designed to be immutable: the data in each line shoul
 
-Snowplow currently supports (or will support in the near future) the following event types:
+Snowplow currently supports the following event types:
 
-| **Event type** | **Value of `event` field in model** |
-| --- | --- |
-| [Page views](#page-views) | 'page_view' |
-| [Page pings](#page-pings) | 'page_ping' |
-| [E-commerce transactions](#e-commerce-transactions) | 'transaction' and 'transaction_item' |
-| [Custom structured events](#custom-structured-events) | 'struct' |
-|[Custom unstructured events](#custom-unstructured-events) | 'unstruct' |
+| Event type                                                | Value of `event` field in model      |
+| --------------------------------------------------------- | ------------------------------------ |
+| [Page views](#page-views)                                 | 'page_view'                          |
+| [Page pings](#page-pings)                                 | 'page_ping'                          |
+| [E-commerce transactions](#e-commerce-transactions)       | 'transaction' and 'transaction_item' |
+| [Custom structured events](#custom-structured-events)     | 'struct'                             |
+| [Custom unstructured events](#custom-unstructured-events) | 'unstruct'                           |
 
-Details of which fields are available for which events are given below.
+Details of which fields are available for which events are given below. In some cases these events will store values in the `atomic.events` columns, which are listed below.
 
 #### Page views
 
+There are currently no fields that are specific to `page_view` events: all the fields that are required are part of the standard fields available for any [web-based event](#web-specific-fields) e.g. `page_urlscheme`, `page_title`.
 There are currently no fields that are specific to `page_view` events: all the fields that are required are part of the standard fields available for any [web-based event](#web-specific-fields) e.g. `page_urlscheme`, `page_title`.
 
 #### Page pings
 
 There are four additional fields included with page pings that indicate how a user has scrolled over a web page since the last page ping:
 
-| **Field** | **Type** | **Description** | **Reqd?** | **Example** |
-| --- | --- | --- | --- | --- |
-| `pp_xoffset_min` | integer | Minimum page x offset seen in the last ping period | No | 0 |
-| `pp_xoffset_max` | integer | Maximum page x offset seen in the last ping period | No | 100 |
-| `pp_yoffset_min` | integer | Minimum page y offset seen in the last ping period | No | 0 |
-| `pp_yoffset_max` | integer | Maximum page y offset seen in the last ping period | No | 200 |
+| Field            | Type    | Description                                        | Reqd? | Example |
+| ---------------- | ------- | -------------------------------------------------- | ----- | ------- |
+| `pp_xoffset_min` | integer | Minimum page x offset seen in the last ping period | No    | 0       |
```

Comment:
More trivia than anything, but I think if the value is 0 it actually isn't populated so you get `null`?

Reply:
Is that just if they are both 0?

Reply:
Nah, if any of the min/max x/y values are falsey they are just not included in the payload. https://github.com/snowplow/snowplow-javascript-tracker/blob/d45e52d404840eaa5745a2f355de722010fccfe9/libraries/tracker-core/src/core.ts#L648

So an actual 0 value for any of these fields is actually unexpected.